scope.Rmd

```diff
@@ -304,102 +379,145 @@ nodes <- nodes %>%
+   assessing and comparing Bayesian and Monte Carlo software in general. (See
+   also the [`MCMCvis`
+   package](https://joss.theoj.org/papers/10.21105/joss.00640), with more of
+   a focus on visualisation.)
+8. The [`walkr` package](https://joss.theoj.org/papers/10.21105/joss.00061) for
+   "MCMC Sampling from Non-Negative Convex Polytopes" is indicative of the
+   difficulties of deriving generally applicable assessments of software in
+   this category, because MCMC *sampling* relies on fundamentally different
+   inputs and outputs than many other MCMC routines.
+
+***Key Considerations*** 
+
+- The extent to which the output of Bayesian routines with uninformative prior inputs can or do
+  reflect equivalent frequentist analyses.
+- Ways to standardise and compare diagnostic statistics for convergence of MCMC
+  routines.
+
+### Dimensionality Reduction and Feature Selection
+
+Many packages either implement or rely upon techniques for dimensionality
+reduction or feature selection. One of the primary problems presented by such
+techniques is that they are constrained to yield a result independent on any
+measure of correctness of accuracy [@estivill-castro_why_2002]. This generally
+precludes any assessment of the accuracy or reliability of such routines.
+Moreover, dimensionality reduction techniques are often developed for
+particular kinds of input data, reducing abilities to compare and contrast
+different implementations, as well as to compare them with any notional
+reference implementations.
+
+
+1. [`ivis`](https://joss.theoj.org/papers/10.21105/joss.01596) implements
+   a dimensionality reduction technique using a "Siamese Neural Network
+   architecture.
+2. [`tsfeaturex`](https://joss.theoj.org/papers/10.21105/joss.01279) is
+   a package to automate "time series feature extraction," which also provides
+   an example of a package for which both input and output data are generally
+   incomparable with most other packages in this category.
+3. [`iRF`](https://joss.theoj.org/papers/10.21105/joss.01077) is another
+   example of a generally incomparable package within this category, here one
+   for which the features extracted are the most distinct predictive features
+   extracted from repeated iterations of random forest algorithms.
+4. [`compboost`](https://joss.theoj.org/papers/10.21105/joss.00967) is
+   a package for component-wise gradient boosting which may be sufficient
+   general to potentially allow general application to problems addressed by
+   several packages in this category. 
+10. The [`iml`](https://joss.theoj.org/papers/10.21105/joss.00786) package may
+    offer usable functionality for devising general assessments of software
+    within this category, through offering a "toolbox for making machine
+    learning models interpretable" in a "model agnostic" way.
+
+***Key Considerations***
+
+- There is generally no way to discern the accuracy of reliability of
+  dimensionality reduction techniques.
```

Comment:
I think this an extreme statement and not exactly accurate? It may be driven by a very specific view of how one might evaluate these things, but I think there are plenty of ways to do so.  A simple example would be k-means, where several algorithms can be shown to reach a certain optimum of minimizing the sum-of-squares of individual clusters.  One could argue that, in failing definitely reach a global optimum rather than local, this fails, but it's a perfectly reasonable criterion that many use and can be tested.  Perhaps a better description is that, unlike regression / supervised ML, it is difficult to use common predictive performance metrics across unsupervised techniques, so evaluation criteria may need to be specific to the method.

Reply:
Yeah, point taken. Although I actually do stand by my primary assertion, it is highly likely to be unnecessarily contentious, and so we'd certainly be better off without it. I'll tone it down accordingly ...